cypress/integration/table.cy.js

```diff
@@ -0,0 +1,113 @@
+import {
+    DIMENSION_ID_PROGRAM_STATUS,
+    DIMENSION_ID_EVENT_STATUS,
+    DIMENSION_ID_CREATED_BY,
+    DIMENSION_ID_LAST_UPDATED_BY,
+    DIMENSION_ID_EVENT_DATE,
+    DIMENSION_ID_ENROLLMENT_DATE,
+    DIMENSION_ID_INCIDENT_DATE,
+    DIMENSION_ID_SCHEDULED_DATE,
+    DIMENSION_ID_LAST_UPDATED,
+} from '../../src/modules/dimensionConstants.js'
+import {
+    ANALYTICS_PROGRAM,
+    TEST_DIM_TEXT,
+    TEST_FIX_PE_DEC_LAST_YEAR,
+} from '../data/index.js'
+import { selectEventProgramDimensions } from '../helpers/dimensions.js'
+import { clickMenubarUpdateButton } from '../helpers/menubar.js'
+import { selectFixedPeriod } from '../helpers/period.js'
+import {
+    getTableRows,
+    getTableHeaderCells,
+    expectTableToBeVisible,
+} from '../helpers/table.js'
+import { EXTENDED_TIMEOUT } from '../support/util.js'
+
+const event = ANALYTICS_PROGRAM
+const dimensionName = TEST_DIM_TEXT
+const periodLabel = event[DIMENSION_ID_EVENT_DATE]
+const testDimensions = [
+    DIMENSION_ID_PROGRAM_STATUS,
+    DIMENSION_ID_EVENT_STATUS,
+    DIMENSION_ID_CREATED_BY,
+    DIMENSION_ID_LAST_UPDATED_BY,
+    DIMENSION_ID_ENROLLMENT_DATE,
+    DIMENSION_ID_INCIDENT_DATE,
+    DIMENSION_ID_SCHEDULED_DATE,
+    DIMENSION_ID_LAST_UPDATED,
+]
+
+const setUpTable = () => {
+    selectEventProgramDimensions({
+        ...event,
+        dimensions: [dimensionName],
+    })
+
+    selectFixedPeriod({
+        label: periodLabel,
+        period: TEST_FIX_PE_DEC_LAST_YEAR,
+    })
+
+    // add main and time dimensions
+    testDimensions.forEach((dimensionId) => {
+        cy.getBySel('main-sidebar')
+            .findBySel(`dimension-item-${dimensionId}`)
+            .find('button')
+            .click({ force: true })
+
+        cy.contains('Add to Columns').click()
+    })
+
+    clickMenubarUpdateButton()
+
+    expectTableToBeVisible()
+
+    cy.getBySelLike('layout-chip').contains(`${dimensionName}: all`)
+}
+
+describe('table', () => {
+    beforeEach(() => {
+        cy.visit('/', EXTENDED_TIMEOUT)
+        setUpTable()
```

Comment:
If there's only 1 test (i.e. `it`) then you might as well skip the `beforeEach` and just do this at the start of the `it` instead.

Reply:
The idea is to add more tests related to table in this file, we have at least sorting and pagination that can be added.

Reply:
Yeah ok but will the other tests be using the same table setup? Seems a bit much to add all of the main dimensions just for sorting and pagination...

Reply:
Those are "problematic" columns because of the remapping going on, so I would assume we do want to test all of them for sorting for example.
Pagination does not have much to do with the table columns.

Reply:
Ok fair enough. Of course feel free to refactor in the future if needed 👍